### PR TITLE
Make PSVersionInfo.PSVersion and PSVersionInfo.PSEdition public

### PR DESCRIPTION
**Pull request by KirkMunro**
## PR Summary

Binary module authors have been neglected, and some data that is easily accessible in PowerShell has not been exposed via public interfaces to binary module authors. I've used reflection in many cases to work around these interface shortcomings since before PowerShell became open source, and I'd like to work away from that and start making these highly important pieces of PowerShell data public so that reflection isn't needed moving forward.

This PR simply makes the static PSVersionInfo.PSVersion and PSVersionInfo.PSEdition properties public so that binary module authors building against PowerShell Core can make runtime decisions based on the version and edition as required without having to invoke PowerShell to retrieve $PSVersionTable and without having to use reflection.

## PR Checklist

- [X] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [X] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [X] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [X] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [X] Issue filed - Issue link: https://github.com/PowerShell/PowerShell-Docs/issues/3127
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Comment by KirkMunro** [comment]
Regarding testing for this feature, I haven't added any automated testing because it was a simple interface visibility change, and I don't believe automated testing is used for that. If automated tests are required, please point me to where you would like them added (as Pester tests, or in C# unit tests). Thank you.

**Comment by iSazonov** [comment]
One note: seems it would be not compatible with Windows PowerShell SDK.

**Comment by KirkMunro** [comment]
@iSazonov How so?

**Comment by iSazonov** [comment]
There could be applications which compiled with both Windows PowerShell SDK and PowerShell Core SDK. It is not big problem in the case - only a note for review.

**Comment by SteveL-MSFT** [comment]
@PowerShell/powershell-committee is ok with this change

File: src/System.Management.Automation/engine/PSVersionInfo.cs [hunk 1]
```diff
@@ -13,9 +13,21 @@
 namespace System.Management.Automation
 {
     /// <summary>
+    /// <para>
     /// Encapsulates $PSVersionTable.
+    /// </para><para>
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/PSVersionInfo.cs]: Please move `<para>` on next new line.

**Review by iSazonov — APPROVED** [review]

**Comment by iSazonov** [comment]
@SteveL-MSFT @adityapatwardhan How we should track adding new public APIs to PowerShell SDK?

**Comment by SteveL-MSFT** [comment]
@iSazonov I created a new label we can use to tag PRs so we can find these easier later

File: src/System.Management.Automation/engine/PSVersionInfo.cs (hunk 1 above)
Comment (KirkMunro) [reply src/System.Management.Automation/engine/PSVersionInfo.cs]: Done

**Review by SteveL-MSFT — APPROVED** [review]

**Pull request merged**